drivers/spi/spi_context.h

```diff
@@ -159,9 +159,12 @@ static inline gpio_flags_t spi_context_cs_active_level(struct spi_context *ctx)
 static inline void spi_context_cs_configure(struct spi_context *ctx)
 {
 	if (ctx->config->cs && ctx->config->cs->gpio_dev) {
-		gpio_flags_t flags = spi_context_cs_active_level(ctx);
+		/* ACTIVE_HIGH & ACTIVE_LOW from DT must be ignored */
+		gpio_dt_flags_t flags = ctx->config->cs->gpio_dt_flags &
+					~GPIO_ACTIVE_LOW;
```

Comment:
I'm going to disagree with @tbursztyka here, and would ask @mnkp to weigh in.

I propose that the active level flags from devicetree should be respected, as a particular signal may pass through an inverter in hardware.  `SPI_CS_ACTIVE`, via `spi_context_cs_{,in}active_value()`, should still be used to determine the value passed to `gpio_pin_set()`.

AFAICT `SPI_CS_ACTIVE_` is only supported with GPIO (in-tree at least) so perhaps it should be deprecated.

Reply:
We used to have a controller that could manage that internally (an intel controller afaik). I am actually surprised to see that many controllers which have CS lines and are thus managing them, do not provide a way for the user to tweak the CS active mode, forcing the user to switch to a gpio line. Let's keep it at it is.

Reply:
`cs-gpios` property attached to spi controller node is there strictly to describe a gpio pin (pin controlled by a GPIO driver) that the SPI driver should use. It cannot be used if the CS pin is to be driven by the SPI module itself. To configure a pin that is driven directly by the SPI module one would have to use pinctrl properties (which we don't support yet).

So in this case we should honor all the supported flags, including `GPIO_ACTIVE_LOW`. At some point `SPI_CS_ACTIVE_` should be deprecated, definitely when we have pinctrl support. And if the following is the case
> AFAICT SPI_CS_ACTIVE_ is only supported with GPIO (in-tree at least) so perhaps it should be deprecated.

we could deprecate `SPI_CS_ACTIVE_` already.

Reply:
> To configure a pin that is driven directly by the SPI module one would have to use pinctrl properties (which we don't support yet).

Thus let's deprecate _once_ we support pinctrl. It does not make sens to deprecate if there is no replacement already available.

Reply:
From what I can tell @mnkp and I agree that the devicetree flags should define the base active level for GPIO control of CS.  @tbursztyka can you accept this?

Whether `SPI_CS_ACTIVE_HIGH` is deprecated can be a separate discussion.  The only in-tree uses of that flag are in drivers that return an error if it's set, so it doesn't seem worthwhile to make it to affect GPIO CS (overriding the devicetree flags), especially as there's agreement to deprecate once pinmux is fully implemented.

So it looks like there are changes to be made to both implementation and documention.  Do we need to bring this to dev-review to try to get agreement on what those changes are?

Reply:
> From what I can tell @mnkp and I agree that the devicetree flags should define the base active level for GPIO control of CS. @tbursztyka can you accept this?

Yes. However SPI_CS_ACTIVE_* should be checked for compliance, so maybe an assert to verify that the gpio dt flags follows the SPI flag would be nice. (@JordanYates could you add this in your PR?)

> Do we need to bring this to dev-review to try to get agreement on what those changes are?
This PR is pretty concise by itself and already gathered relevant stakeholders (reviewers list).


Let's bring the deprecation of SPI_CS_ACTIVE_* flags when there will be a pinctrl API. That's unrelated to that PR.

Reply:
CS active level is now taken from devicetree, with checks that it is equivalent to the SPI_CS_ACTIVE_* value.